### PR TITLE
BUG: Fix errors for WSL machines with GPU.

### PR DESCRIPTION
**Pull request by yifeis7**
## What do these changes do?

Fix errors for WSL machines with GPU. 
Machine configuration:
CPU: i7-12700
GPU: RTX 3060 Laptop
### How to reproduce
Run the following code snippets in WSL(Ubuntu 22.04.2)
```
import xorbits.numpy as np
import xorbits.pandas as pd
size = 100000
df = pd.DataFrame(np.random.rand(size, 4), columns=list('abcd'))
print(df.sum())
```
<details>
<summary>Bug stack trace</summary>
/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py:1873: UserWarning: No existing session found, creating a new local session now.
  warnings.warn(warning_msg)
2023-06-06 11:55:44,554 xorbits._mars.services.cluster.uploader 5537 ERROR    Failed to upload node info
Traceback (most recent call last):
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/uploader.py", line 133, in upload_node_info
    self._info.env = await asyncio.to_thread(gather_node_env)
  File "/home/yifei/anaconda3/lib/python3.9/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/home/yifei/anaconda3/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/gather.py", line 128, in gather_node_env
    mars_resource.cuda_card_stats()
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/resource.py", line 397, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/lib/nvutils.py", line 498, in get_device_status
    gpu_util = int(util_node.find("gpu_util"))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'xml.etree.ElementTree.Element'
2023-06-06 11:55:44,555 xorbits._mars.services.cluster.uploader 5537 ERROR    Failed to upload node info: int() argument must be a string, a bytes-like object or a number, not 'xml.etree.ElementTree.Element'
Traceback (most recent call last):
  File "/home/yifei/codeVM/test/learn/learn.py", line 3, in <module>
    import xorbits.numpy as np
  File "/home/yifei/codeVM/xorbits/python/xorbits/utils.py", line 37, in inn
    return f(self, *args, **kwargs)
  File "/home/yifei/codeVM/xorbits/python/xorbits/core/data.py", line 277, in __str__
    run(self)
  File "/home/yifei/codeVM/xorbits/python/xorbits/core/execution.py", line 55, in run
    mars_execute(mars_tileables, **kwargs)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1707, in execute
    session = get_default_or_create(**(new_session_kwargs or dict()))
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1874, in get_default_or_create
    session = new_session("127.0.0.1", init_local=True, **kwargs)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1844, in new_session
    session = SyncSession.init(
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1476, in init
    isolated_session = fut.result()
  File "/home/yifei/anaconda3/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/home/yifei/anaconda3/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 742, in init
    await new_cluster_in_isolation(
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/local.py", line 95, in new_cluster_in_isolation
    await cluster.start()
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/local.py", line 302, in start
    await self._start_service()
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/local.py", line 357, in _start_service
    self._web = await start_supervisor(
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/deploy/oscar/service.py", line 45, in start_supervisor
    await start_services(NodeRole.SUPERVISOR, config, address=address)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/core.py", line 175, in start_services
    await asyncio.gather(*[inst.start() for inst in instances])
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/supervisor/service.py", line 65, in start
    await mo.create_actor(
  File "/home/yifei/anaconda3/lib/python3.9/site-packages/xoscar/api.py", line 30, in create_actor
    return await ctx.create_actor(actor_cls, *args, uid=uid, address=address, **kwargs)
  File "/home/yifei/anaconda3/lib/python3.9/site-packages/xoscar/backends/context.py", line 120, in create_actor
    return self._process_result_message(result)
  File "/home/yifei/anaconda3/lib/python3.9/site-packages/xoscar/backends/context.py", line 79, in _process_result_message
    raise message.as_instanceof_cause()
  File "/home/yifei/anaconda3/lib/python3.9/site-packages/xoscar/backends/pool.py", line 543, in create_actor
    await self._run_coro(message.message_id, actor.__post_create__())
  File "/home/yifei/anaconda3/lib/python3.9/site-packages/xoscar/backends/pool.py", line 360, in _run_coro
    return await coro
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/uploader.py", line 62, in __post_create__
    await self._uploaded_future
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/uploader.py", line 88, in _periodical_upload_node_info
    await self.upload_node_info()
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/uploader.py", line 133, in upload_node_info
    self._info.env = await asyncio.to_thread(gather_node_env)
  File "/home/yifei/anaconda3/lib/python3.9/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/home/yifei/anaconda3/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/services/cluster/gather.py", line 128, in gather_node_env
    mars_resource.cuda_card_stats()
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/resource.py", line 397, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "/home/yifei/codeVM/xorbits/python/xorbits/_mars/lib/nvutils.py", line 498, in get_device_status
    gpu_util = int(util_node.find("gpu_util"))
TypeError: [address=127.0.0.1:40565, pid=5537] int() argument must be a string, a bytes-like object or a number, not 'xml.etree.ElementTree.Element'
</details>

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/lib/nvutils.py [hunk 1]
```diff
@@ -494,11 +494,17 @@ def get_device_status(dev_index: int) -> _nvml_device_status:
         if util_node.find("gpu_util").text == "N/A":
             gpu_util = 0
         else:
-            gpu_util = int(util_node.find("gpu_util"))
+            try:
+                gpu_util = int(util_node.find("gpu_util"))
+            except:
```
Comment (UranusSeven) [inline python/xorbits/_mars/lib/nvutils.py:499]: Better catch an explicit exception and explain in what scenario this exception happen.

File: python/xorbits/_mars/lib/nvutils.py [hunk 2]
```diff
@@ -494,11 +494,17 @@ def get_device_status(dev_index: int) -> _nvml_device_status:
...
+                gpu_util = int(util_node.find("gpu_util").text.split()[0])
         if util_node.find("memory_util").text == "N/A":
             mem_util = 0
         else:
-            mem_util = int(util_node.find("memory_util"))
+            try:
+                mem_util = int(util_node.find("memory_util"))
+            except:
```
Comment (UranusSeven) [inline python/xorbits/_mars/lib/nvutils.py:506]: Ditto.

File: python/xorbits/_mars/resource.py [hunk 3]
```diff
@@ -393,22 +393,24 @@ def cuda_card_stats() -> List[_cuda_card_stat]:  # pragma: no cover
...
+                        total=device_status.fb_total_mem,
+                        used=device_status.fb_used_mem,
+                        free=device_status.fb_free_mem,
+                        available=device_status.fb_free_mem,
+                        percent=device_status.mem_util,
+                    ),
+                )
             )
-        )
+        except:
```
Comment (UranusSeven) [inline python/xorbits/_mars/resource.py:414]: Ditto.

File: python/xorbits/_mars/lib/nvutils.py (hunk 1 above)
Comment (UranusSeven) [reply python/xorbits/_mars/lib/nvutils.py:499]: Or you may print some logs when exceptions happen, which could help to learn what happened.

Comment (yifeis7) [reply python/xorbits/_mars/lib/nvutils.py:499]: Got it.

File: python/xorbits/_mars/lib/nvutils.py [hunk 4]
```diff
@@ -494,12 +494,23 @@ def get_device_status(dev_index: int) -> _nvml_device_status:
         if util_node.find("gpu_util").text == "N/A":
             gpu_util = 0
         else:
-            gpu_util = int(util_node.find("gpu_util"))
+            try:
+                gpu_util = int(util_node.find("gpu_util"))
+            except TypeError:
+                # Solve the problem of illegal conversion from string to int
+                gpu_util = int(util_node.find("gpu_util").text.split()[0])
```
Comment (UranusSeven) [inline python/xorbits/_mars/lib/nvutils.py:501]: What if line 501 raised an error?

File: python/xorbits/_mars/lib/nvutils.py [hunk 5]
```diff
@@ -494,12 +494,23 @@ def get_device_status(dev_index: int) -> _nvml_device_status:
...
         if util_node.find("memory_util").text == "N/A":
             mem_util = 0
         else:
-            mem_util = int(util_node.find("memory_util"))
-
+            try:
+                mem_util = int(util_node.find("memory_util"))
+            except TypeError:
+                # Solve the problem of illegal conversion from string to int
+                mem_util = int(util_node.find("memory_util").text.split()[0])
```
Comment (UranusSeven) [inline python/xorbits/_mars/lib/nvutils.py:511]: Ditto.

File: python/xorbits/_mars/resource.py [hunk 6]
```diff
@@ -393,22 +393,24 @@ def cuda_card_stats() -> List[_cuda_card_stat]:  # pragma: no cover
...
+                        used=device_status.fb_used_mem,
+                        free=device_status.fb_free_mem,
+                        available=device_status.fb_free_mem,
+                        percent=device_status.mem_util,
+                    ),
+                )
             )
-        )
+        except Exception as e:
+            logger.warning(f"Caught exception of type: {type(e).__name__}")
```
Comment (UranusSeven) [inline python/xorbits/_mars/resource.py:415]:
```suggestion
            logger.warning(f"Unexpected error happened when gathering stats of device {device_idx}", exc_info=True)
```

File: python/xorbits/_mars/lib/nvutils.py (hunk 4 above)
Comment (yifeis7) [reply python/xorbits/_mars/lib/nvutils.py]: I integrated them into the "try" branch.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by UranusSeven — APPROVED** [review]
LGTM.

**Review by aresnow1 — APPROVED** [review]

**Pull request merged**